src/marginaldamage.jl

```diff
@@ -61,25 +61,25 @@ get_marginal_model(m::Model = get_model(); year::Int = nothing)
 Creates a Mimi MarginalModel where the provided m is the base model, and the marginal model has additional emissions of CO2 in year `year`.
 If no Model m is provided, the default model from MimiDICE2010.get_model() is used as the base model.
 """
-function get_marginal_model(m::Model=get_model(); year::Union{Int, Nothing} = nothing)
+function get_marginal_model(m::Model=get_model(); year::Union{Int, Nothing} = nothing, pulse_size::Float64=1e10)
     year === nothing ? error("Must specify an emission year. Try `get_marginal_model(m, year=2015)`.") : nothing
     !(year in model_years) ? error("Cannot add marginal emissions in $year, year must be within the model's time index $(model_years[1]):10:$last_year.") : nothing
 
-    mm = create_marginal_model(m, 1e10) # Pulse has a value of 1GtC per year for ten years
-    add_marginal_emissions!(mm.marginal, year)
+    mm = create_marginal_model(m, pulse_size) # Pulse has a value of 1GtC per year for ten years
+    add_marginal_emissions!(mm.marginal, year, pulse_size)
 
     return mm
 end
 
 """
 Adds a marginal emission component to year m which adds 1Gt of additional C emissions per year for ten years starting in the specified `year`.
 """
-function add_marginal_emissions!(m::Model, year::Int) 
+function add_marginal_emissions!(m::Model, year::Int, pulse_size::Float64) 
     add_comp!(m, Mimi.adder, :marginalemission, before=:co2cycle)
 
     time = Mimi.dimension(m, :time)
     addem = zeros(length(time))
-    addem[time[year]] = 1.0     # 1 GtC per year for ten years
+    addem[time[year]] = pulse_size / 1e10     # Unit of pulse_size is in GtC, we convert to ton by dividing by 1e9, and then divide by 10 again because that impulse is emitted for ten years
```

Comment:
change comment to: "Unit of pulse_size is tons, but units of emissions in DICE are GtC, so we convert to GtC by dividing by 1e9, and then divide by 10 again because that pulse is emitted for ten years."